datumaro/plugins/imagenet_format.py

```diff
@@ -59,14 +83,24 @@ def apply(self):
         subset_dir = self._save_dir
         extractor = self._extractor
         image_dirs = {}
+        labels = {}
         for subset in self._extractor.subsets().items():
             for item in subset[1]:
-                image_name = self._make_image_filename(item)
-                if item.annotations[0].type == AnnotationType.label:
-                    label = item.annotations[0].label
-                    if self._save_images:
-                        if label not in image_dirs:
-                            image_dirs[label] = osp.join(subset_dir,
-                                extractor.categories()[AnnotationType.label][label].name)
-                            os.makedirs(image_dirs[label], exist_ok=True)
-                        self._save_image(item, osp.join(image_dirs[label], image_name))
+                image_name = item.id
+                labels[image_name] =  [p.label for p in item.annotations]
+                for label in labels[image_name]:
+                    label_name = extractor.categories()[AnnotationType.label][label].name
+                    if label not in image_dirs:
+                        image_dirs[label] = osp.join(subset_dir, label_name)
+                        os.makedirs(image_dirs[label], exist_ok=True)
+                    self._save_image(item, osp.join(image_dirs[label],
```

Comment:
There is a parameter for directory creation in `save_image()`, so missing dirs will be created automatically with base `_save_image()`. It will also cover multiple dirs from image ids.